particles/field.h

```diff
@@ -60,79 +58,157 @@ struct Field
       }
    }
 
-   Vec3d getCell(int x, int y, int z) {
+   std::vector<double> getCell(int x, int y, int z) {
```

Comment:
I would suggest that this be ```std::array<double, 3>```. Because why would you use a variable-sized vector here?